subprojects/testing-native/src/integTest/groovy/org/gradle/nativeplatform/test/AbstractNativeUnitTestIntegrationTest.groovy

```diff
@@ -16,9 +16,9 @@
 
 package org.gradle.nativeplatform.test
 
-import org.gradle.language.AbstractNativeLanguageComponentIntegrationTest
+import org.gradle.nativeplatform.fixtures.AbstractInstalledToolChainIntegrationSpec
 
-abstract class AbstractNativeUnitTestIntegrationTest extends AbstractNativeLanguageComponentIntegrationTest {
+abstract class AbstractNativeUnitTestIntegrationTest extends AbstractInstalledToolChainIntegrationSpec {
```

Comment:
you need to duplicate each of the subclasses of this, so that one inherits from `AbstractNativeLanguageComponentIntegrationTest` and the other inherits from `AbstractNativeUnitTestIntegrationTest`. Otherwise, we've lost a whole pile of coverage that used to be inherited from `AbstractNativeLanguageComponentIntegrationTest`.

Reply:
I will duplicate the test class, but after giving it some thought it seems more that `AbstractNativeLanguageComponentIntegrationTest` and `AbstractSwiftComponentIntegrationTest` were more focus on the component itself. In the XCTest, it's the `XCTestSuite` that we want to test with `AbstractSwiftComponentIntegrationTest` as opposed to `AbstractNativeUnitTestIntegrationTest` where we test that it can behave with a tested component. I may be wrong but it seems we would be duplicating coverage that shouldn't be needed. However, you are right that is we want to be 110% certain of the coverage of every feature, we should duplicate it. I will add the classes.

Reply:
There wouldn't be any duplication. The tests that verify that the test suite component behaves well as a component can go in one hierarchy, and the tests that verify that the test suite behaves well as a (unit) test suite can go in another hierarchy. They wouldn't share a root class with any tests in it.